srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/CalculationResult.java

```diff
@@ -0,0 +1,216 @@
+package gov.va.med.srcalc.domain.calculation;
+
+import gov.va.med.srcalc.domain.Patient;
+import gov.va.med.srcalc.domain.model.DisplayNameComparator;
+import gov.va.med.srcalc.domain.model.Specialty;
+
+import java.io.Serializable;
+import java.util.*;
+
+import org.joda.time.DateTime;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.*;
+
+/**
+ * <p>A calculation result. Stores both the calculation input values and the
+ * outcomes. Immutable as long as the provided {@link Value} objects are
+ * immutable.</p>
+ * 
+ * <p>Per Effective Java Item 17, this class is marked final because it was not
+ * designed for inheritance.</p>
+ */
+public final class CalculationResult implements Serializable
+{
+    /**
+     * Change this when changing the class!
+     */
+    private static final long serialVersionUID = 2L;
+    
+    private static final Comparator<Value> VALUE_COMPARATOR =
+            new ValueVariableComparator(new DisplayNameComparator());
+
+    private final DateTime fStartDateTime;
+    private final int fPatientDfn;
+    private final String fSpecialtyName;
+    private final Optional<ProcedureValue> fProcedureValue;
+    /**
+     * The non-ProcedureValue values. Store it sorted by display name since we
+     * need that order for {@link #buildNoteBody()}.
+     */
+    private final ImmutableSortedSet<Value> fNonProcedureValues;
+    private final ImmutableSortedMap<String, Double> fOutcomes;
+    
+    /**
+     * <p>Constructs an instance with the given properties.</p>
+     * 
+     * <p>Note: does not perform any consistency check on the data. For example,
+     * this constructor will accept outcomes for models not in the given
+     * specialty.</p>
+     * 
+     * @param startDateTime when the calculation was started, for metrics
+     * @param patientDfn the associated patient DFN
+     * @param specialtyName the associated specialty name, must not be null
+     * @param values the calculation input values, must not be null. The passed
+     * collection may be mutable as this object will take an immutable snapshot.
+     * @param outcomes the calculation outcomes, must not be null. The passed
+     * collection may be mutable as this object will take an immutable snapshot.
+     * @throws NullPointerException if any argument is null or any value in a
+     * collection argument is null
+     * @throws IllegalArgumentException if the given values set contains multiple
+     * ProcedureValues
+     */
+    public CalculationResult(
+            final DateTime startDateTime,
+            final int patientDfn,
+            final String specialtyName,
+            final Set<Value> values,
+            final Map<String, Double> outcomes)
+    {
+        fStartDateTime = startDateTime;
+        fPatientDfn = patientDfn;
+        fSpecialtyName = Objects.requireNonNull(
+                specialtyName, "specialty name must be non-null");
```

Comment:
We have a limit in the database creation that specifies specialties are 255 characters, but what about in Java code?